pkg/register/register.go

```diff
@@ -100,11 +102,29 @@ func Register(url string, caCert []byte, smbios bool, emulateTPM bool, emulatedS
 	if err := WriteMessage(conn, MsgGet, []byte{}); err != nil {
 		return nil, fmt.Errorf("request elemental configuration: %w", err)
 	}
-	_, r, err := conn.NextReader()
+
+	msgType, data, err := ReadMessage(conn)
 	if err != nil {
-		return nil, fmt.Errorf("read elemental configuration: %w", err)
+		return nil, fmt.Errorf("read configuration response: %w", err)
+	}
+
+	logrus.Debugf("Got configuration response: %s", msgType.String())
+
+	switch msgType {
+	case MsgError:
+		msg := &ErrorMessage{}
+		if err = yaml.Unmarshal(data, &msg); err != nil {
+			return nil, errors.Wrap(err, "unable to unmarshal error-message")
+		}
+
+		return nil, errors.New(msg.Message)
+
+	case MsgConfig:
+		return data, nil
+
+	default:
+		return nil, fmt.Errorf("Unknown response message-type: %s", msgType)
```

Comment:
nit: here you will get either an unexpected known message (that will expand to the corresponding string) or a code not know that will expand to "Unknown".
```suggestion
		return nil, fmt.Errorf("unexpected response message: %s", msgType)
```